flocker/control/httpapi.py

```diff
@@ -53,6 +68,116 @@ def version(self):
         """
         return {u"flocker":  __version__}
 
+    @app.route("/datasets", methods=['POST'])
+    @user_documentation(
+        """
+        Create a new dataset.
+        """,
+        examples=[
+            u"create dataset",
+            u"create dataset with dataset_id",
+            u"create dataset with duplicate dataset_id",
+            u"create dataset with maximum_size",
+            u"create dataset with metadata",
+        ]
+    )
+    @structured(
+        inputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets'},
+        outputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets'},
+        schema_store=SCHEMAS
+    )
+    def create_dataset(self, primary, dataset_id=None, maximum_size=None,
+                       metadata=None):
+        """
+        Create a new dataset in the cluster configuration.
+
+        :param unicode primary: The address of the node on which the primary
+            manifestation of the dataset will be created.
+
+        :param unicode dataset_id: A unique identifier to assign to the
+            dataset.  This is expected to be a UUID.  If no value is given, one
```

Comment:
"expected to be a UUID" is a little vague, there's a few alternate formats I believe.